my-sms/spec/factories/students.rb

```diff
@@ -2,7 +2,7 @@
 
 FactoryBot.define do
   factory :student do
-    title { %w[Mr Mrs Miss Ms].sample }
+    association :title, strategy: :build
```

Comment:
Probably want the default to just be the create strategy. I think in this case it would be okay to leave as the default

```suggestion
    title
```

Reply:
I use the build strategy because there's a standard set of titles created by the migration. This creates a conflict when creating the record because there is a uniqueness constraint on the title name.

I could replace this with something that won't appear in the database (e.g. title1, title2, title3,...) which would enable the create strategy to be used at the expense of readability.

Reply:
In this case I don't think it will be an issue since we're no longer adding the data during the migration. Usually we would use a tool like [database_cleaner](https://github.com/DatabaseCleaner/database_cleaner) to clean the database inbetween test invocations anyways. You might need to just manually reset your test db otherwise for now just allow creation of dummy titles (like `test1`, `test2`, etc..)